### DISCUSSION TITLE
How do I copy the keys for optional allowed mods?

### DISCUSSION THREAD

**Discussion by mwpowellhtx** [Foxlider/FASTER#148] [Q&A]
i.e. Such that the server should not need to load them, but are _approved_, _allowed_, _client side_? things like _CH View Distance_, _DUI - Squad Radar_, and the like. did not see a mention of it in the docs, or perhaps it is obscure... the mods are available via the imported mod set, downloaded and such.

I tried the _Copy Mods Keys_, but this does not seem to do anything, nothing to _clipboard_, no _files copied_ i.e. via _Windows Explorer_, nothing like that?

Is it sufficient to _deploy_ them even though they are not among the server side `-mod` or `-serverMod`?

**Comment by OverlordZorn** [comment]
Optional Mods need the be handled manually as of now. There is a PR that would add a optional collumn but that isnt live yet.

How do you handle optional mods?
You need to move the key from the mod per hand in the keys folder while not loading it on the server.(and the client, for that matter (optional and so on)).

There is the option to load an optional mod *also* as a `server only mod` via the `server` tab to enforce some cba settings if needed.


Deployment in Faster is simply a syslink from the Mod Staging folder to the ArmaServer Root folder and has not relevance regarding what mods are required, but it is required for any mod to be used on the server. simply make sure when adding new mods to always deploy all of them.

This method of Deployment allows faster to be used for multiple servers while using same ModStagingFolder but different server instances, different modlists etc.

**Reply by mwpowellhtx** [reply]
Understood, thanks for clarifying.

I think that's more or less what I ended up doing, mapped from the ID to the folder and copied them over 1x1.

Yep, also, figured out the `-serverMod` aspect, key for things like _Advanced Towing_, very handy, that.

**Reply by OverlordZorn** [reply]
discord has a channel/forum with a collection of potentially useful server only mods - worth checking out

**Reply by mwpowellhtx** [reply]
That's good to know as well, perhaps a few more could be server only. That helps, thanks...

**Reply by OverlordZorn** [reply]
i added a few lines for clarity in the original reply.

**Reply by mwpowellhtx** [reply]
> discord has a channel/forum with a collection of potentially useful server only mods - worth checking out

do you have a message link to that, perhaps? which assuming by _discord_ you meant the _FASTER_ server.

**Reply by OverlordZorn** [reply]
https://discord.com/channels/366955806777671681/1034538130310709359

**Comment by mwpowellhtx** [comment]
Actually, kinda dawned on me, don't **_ALL_** the keys need to be copied into `.\keys`? Which I suppose is also the selling point of a `.\staging` area, could just search `*.bikey`, aye?
